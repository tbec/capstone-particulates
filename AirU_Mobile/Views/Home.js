import React, {Component} from 'react';
import {Text, View, TouchableOpacity, Image} from 'react-native';
import {TabNavigator} from '../Components/Navigation'
import {NavigationActions} from 'react-navigation'

export default class Home extends Component {
    // need way to check if already setup or if this is first time, 
    // Pass into home? - TC
    
    render() {
        var styles = require('../StyleSheets/Styles');
        return (
            <View style={styles.container}>
                <Text>Home Screen</Text>
                <TouchableOpacity onPress={() => this.props.navigation.navigate({
                    routeName: 'Tabs',
                    params: {},
                    action: NavigationActions.navigate({routeName: 'Home'})})}>
                    <Image source={require("../Resources/Sensor.png")}/>
                </TouchableOpacity>
            </View>
        );
    }
}