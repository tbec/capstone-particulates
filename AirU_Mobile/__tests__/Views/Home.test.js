import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import Home from '../../Views/Home'

// used to find testIDs in code
let findById = function(tree, testID) {
    if(tree.props && tree.props.testID === testID) {
        return tree
    }
    if(tree.children && tree.children.length > 0)
    {
        let childs = tree.children
        for(let i = 0; i < childs.length; i++)
        {
            let item = findById(childs[i], testID)
            if(typeof(item) !== 'undefined') {
                return item
            }
        }
    }
}

// create snapshot for home screen
test('Home Snapshot', () => {
    const snap = renderer.create(<Home/>).toJSON();
    expect(snap).toMatchSnapshot();
})

// check sensor
test('Select Sensor', () => {
    const home = renderer.create(<Home/>).toJSON();
    button = findById(home, 'SensorButton')
    expect(button).toBeDefined();
})

// check Tracker
test('Select Tracker', () => {
    const home = renderer.create(<Home/>).toJSON();
    button = findById(home, 'TrackerButton')
    expect(button).toBeDefined();
})

// check Map
test('Select Map', () => {
    const home = renderer.create(<Home/>).toJSON();
    button = findById(home, 'MapButton')
    expect(button).toBeDefined();
})

// check Settings
test('Select Settings', () => {
    const home = renderer.create(<Home/>).toJSON();
    button = findById(home, 'SettingsButton')
    expect(button).toBeDefined();
})