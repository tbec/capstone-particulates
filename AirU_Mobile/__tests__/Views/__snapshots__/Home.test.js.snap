// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../Resources/home_background.jpg",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": "100%",
            "width": "100%",
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 5,
          "flexDirection": "column",
          "padding": 40,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../Resources/AQ_Logo.png",
          }
        }
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 100,
            "flexDirection": "column",
            "justifyContent": "space-evenly",
          },
          Object {
            "flex": 100,
          },
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "red",
            "borderColor": "black",
            "borderWidth": 2,
            "height": 50,
            "justifyContent": "space-evenly",
            "padding": 1,
            "width": 150,
          }
        }
        testID="SensorButton"
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontWeight": "bold",
            }
          }
        >
          Sensor
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "red",
            "borderColor": "black",
            "borderWidth": 2,
            "height": 50,
            "justifyContent": "space-evenly",
            "padding": 1,
            "width": 150,
          }
        }
        testID="TrackerButton"
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontWeight": "bold",
            }
          }
        >
          Personal Tracker
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "red",
            "borderColor": "black",
            "borderWidth": 2,
            "height": 50,
            "justifyContent": "space-evenly",
            "padding": 1,
            "width": 150,
          }
        }
        testID="MapButton"
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontWeight": "bold",
            }
          }
        >
          Sensor Map
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "red",
            "borderColor": "black",
            "borderWidth": 2,
            "height": 50,
            "justifyContent": "space-evenly",
            "padding": 1,
            "width": 150,
          }
        }
        testID="SettingsButton"
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontWeight": "bold",
            }
          }
        >
          Settings
        </Text>
      </View>
    </View>
  </View>
</View>
`;
